File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    n = len(keymap)
    dic = {}
    for i in range(n):
        temp = list(keymap[i])
        for j in range(len(temp)):
            if temp[j] not in dic:
                dic[temp[j]] = j+1
            else:
                dic[temp[j]] = min(j+1, dic[temp[j]])
                
    for target in targets:
        cnt = 0
        for c in target:
            if c in dic: cnt += dic[c]
            else:
                cnt = -1
                break
        answer.append(cnt)
    return answer

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        int cnt;
        Map<Character, Integer> map = new HashMap<>();
        
        for (String keylist: keymap)
            for (int i=0;i < keylist.length();i++)
                map.put(keylist.charAt(i), Math.min(map.getOrDefault(keylist.charAt(i), keylist.length()), i + 1));
        
        for (int i=0;i < targets.length;i++) {
            cnt = 0;
            for (char c: targets[i].toCharArray()) {
                if (map.containsKey(c))
                    cnt += map.get(c);
                else {
                    cnt = -1;
                    break;
                }
            }
            answer[i] = cnt;        
        }
        return answer;
    }
}